airflow/www/static/js/tree/details/content/taskInstance/taskActions/Clear.jsx

```diff
@@ -0,0 +1,127 @@
+/*!
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import React, { useState } from 'react';
+import {
+  Button,
+  Flex,
+  ButtonGroup,
+  useDisclosure,
+} from '@chakra-ui/react';
+
+import ActionButton from './ActionButton';
+import ConfirmDialog from '../../ConfirmDialog';
+import { useClearTask, useConfirmClearTask } from '../../../../api';
+
+const Run = ({
+  dagId,
+  runId,
+  taskId,
+  executionDate,
+}) => {
+  const [affectedTasks, setAffectedTasks] = useState([]);
+
+  // Options check/unchecked
+  const { isOpen: past, onToggle: onTogglePast } = useDisclosure();
```

Comment:
Seems a little unorthodox to use `useDisclosure` to manage these boolean state values, no?